contracts/src/emergency/sequencer_uptime_feed.cairo

```diff
@@ -0,0 +1,305 @@
+
+    #[event]
+    fn UpdateIgnored(
+        latest_status: u128, latest_timestamp: u64, incoming_status: u128, incoming_timestamp: u64
+    ) {}
+
+    #[event]
+    fn L1SenderTransferred(from_address: felt252, to_address: felt252) {}
+
+    impl Aggregator of IAggregator {
+        fn latest_round_data() -> Round {
+            require_access();
+            let latest_round_id = _latest_round_id::read();
+            let round_transmission = _round_transmissions::read(latest_round_id);
+            Round {
+                round_id: latest_round_id.into(),
+                answer: round_transmission.answer,
+                block_num: round_transmission.block_num,
+                started_at: round_transmission.observation_timestamp,
+                updated_at: round_transmission.transmission_timestamp,
+            }
+        }
+
+        fn round_data(round_id: u128) -> Round {
+            require_access();
+            assert(round_id < _latest_round_id::read(), 'invalid round id');
+            let round_transmission = _round_transmissions::read(round_id);
+            Round {
+                round_id: round_id.into(),
+                answer: round_transmission.answer,
+                block_num: round_transmission.block_num,
+                started_at: round_transmission.observation_timestamp,
+                updated_at: round_transmission.transmission_timestamp,
+            }
+        }
+
+        fn description() -> felt252 {
+            'L2 Sequencer Uptime Status Feed'
+        }
+
+        fn decimals() -> u8 {
+            0_u8
+        }
+
+        fn type_and_version() -> felt252 {
+            'SequencerUptimeFeed 1.0.0'
+        }
+    }
+
+    #[constructor]
+    fn constructor(initial_status: u128, owner_address: ContractAddress) {
+        initializer(initial_status, owner_address);
+    }
+
+    #[l1_handler]
+    fn update_status(from_address: felt252, status: u128, timestamp: u64) {
+        assert(_l1_sender::read() == from_address, 'EXPECTED_FROM_BRIDGE_ONLY');
+
+        let latest_round_id = _latest_round_id::read();
+        let latest_round = _round_transmissions::read(latest_round_id);
+
+        if timestamp <= latest_round.observation_timestamp {
+            UpdateIgnored(
+                latest_round.answer, latest_round.transmission_timestamp, status, timestamp
+            );
+            return ();
+        }
+
+        if latest_round.answer == status {
+            _update_round(latest_round_id, latest_round);
+        } else {
+            // only increment round when status flips
+            let round_id = latest_round_id + 1_u128;
+            _record_round(round_id, status, timestamp);
+        }
+    }
+
+    #[external]
+    fn set_l1_sender(address: felt252) {
+        Ownable::assert_only_owner();
+
+        // convert both to u256 (felts don't implement PartialOrd)
+        assert(address.into() < ETH_ADDRESS_BOUND.into(), 'invalid eth address');
```

Comment:
Is this correct? I thought the ETH addresses are above a certain bound (have to be long enough)

Reply:
Eth addresses are 160 bytes, so if I'm not mistaken we just need to check if the address type is lower than the highest possible value. 

in the cairo token_bridge examples [they do something simila](https://github.com/starkware-libs/cairo/blob/main/crates/cairo-lang-starknet/test_data/token_bridge.cairo#L27)r (comparing two u256s)  so I think we should be golden 🏅